main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -3304,29 +3317,25 @@ object Resolver {
     //
     // The enum is not accessible.
     //
-    Validation.toHardFailure(ResolutionError.InaccessibleRestrictableEnum(enum0.sym, ns0, loc))
+    Validation.toHardFailure(ResolutionError.InaccessibleRestrictableEnum(enum0.sym, ns0, loc)) // TODO: Refactor this to be soft failure
```

Comment:
I think this can be a soft failure since everything else is. It may require some handling in later phases that assume this is not possible but that should more or less just be the same as the enum case. Also, there is no test for this error, so I think we should add that first :)

Reply:
This can def be a soft error